yroots/IntervalChecks.py

```diff
@@ -117,24 +156,25 @@ def check_subintervals(self, subintervals, scaled_subintervals, polys, change_si
             The MultiCheb polynomials that approximate the functions on these intervals.
         change_sign: list
             A list of bools of whether we know the functions can change sign on the subintervals.
-        approx_tol: float
-            The sup norm bound on the approximation error.
+        errors: list
+            The approximation errors of the polynomials.
         Returns
         -------
         check_interval : bool
             True if we can throw out the interval. Otherwise False.
         '''
         for check in self.subinterval_checks:
-            for poly in polys:
-                mask = check(poly, scaled_subintervals, change_sign, approx_tol)
+            for poly,error in zip(polys, errors):
+                mask = check(poly, scaled_subintervals, change_sign, error)
```

Comment:
I replaced approx_tol everywhere with the actual approximation error as measured earlier. This is the sum of the absolute values of the coefficients that are removed from the degree 2n approximation to get our approximation